src/aap_eda/core/management/commands/purge_log_records.py

```diff
@@ -0,0 +1,139 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from datetime import datetime
+from typing import Union
+
+from django.core.management.base import BaseCommand, CommandParser
+from django.db import transaction
+from django.db.models import Q
+
+from aap_eda.core import models
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Purge the logs from a rulebook process."""
+
+    help = (
+        "Purge log records from rulebook processes. "
+        "If activation ids or names are not specified, "
+        "all log records older than the cutoff date will be purged."
+    )
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            "--activation-ids",
+            nargs="+",
+            type=int,
+            dest="activation-ids",
+            help=(
+                "Specify the activation ids which you want to clear their "
+                "records (e.g., ActivationID1 ActivationID2)"
+            ),
+        )
+        parser.add_argument(
+            "--activation-names",
+            nargs="+",
+            type=str,
+            dest="activation-names",
+            help=(
+                "Specify the activation names which you want to clear their "
+                "records (e.g., ActivationName1 ActivationName2)"
+            ),
+        )
+        parser.add_argument(
+            "--date",
+            dest="date",
+            action="store",
+            required=True,
+            help=(
+                "Purge records older than this date from the database. "
+                "The cutoff date in YYYY-MM-DD format"
+            ),
+        )
+
+    def purge_log_records(
+        self, ids: list[int], names: list[str], cutoff_date: str
+    ) -> None:
+        new_instance_logs = []
+        if not bool(ids) and not bool(names):
+            instances = models.RulebookProcess.objects.all()
+        else:
+            instances = models.RulebookProcess.objects.filter(
+                Q(activation__id__in=ids) | Q(activation__name__in=names),
+            )
+
+        if not instances.exists():
+            logger.info("No records has been found for purging.")
```

Comment:
This should not be a logger:
```suggestion
            self.stdout.write(
                self.style.SUCCESS("No records has been found for purging.")
            )
```